### PR TITLE
Clean ups to the swift-syntax-dev-utils and Package.swift

### PR DESCRIPTION
**Pull request by ahoppen**
- Move the environment variables that we check for in Package.swift to the top and add documentation to them
- Perform some clean-ups in Package.swift
  - Don’t set `SWIFTSYNTAX_NO_OSLOG_DEPENDENCY` because it’s not needed
  - Use the Swift settings of a target also for the corresponding test target. That’s what we already did for SwiftParser and SwiftParserTests. We should be consistent here.
- Don’t set `--verbose` twice when invoking SwiftPM from `swift-syntax-dev-utils`
  - `invokeSwiftPM` already sets `--verbose`. No need to pass it again in the `additionalArguments`.
- Remove `--disable-sandbox`. This flag wasn’t used anywhere.

### CODE REVIEW

File: CodeGeneration/Tests/ValidateSyntaxNodes/ValidateSyntaxNodes.swift [hunk 1]
```diff
@@ -515,6 +515,8 @@ class ValidateSyntaxNodes: XCTestCase {
         ValidationFailure(node: .differentiabilityArguments, message: "could conform to trait 'Parenthesized' but does not"),
         ValidationFailure(node: .editorPlaceholderDecl, message: "could conform to trait 'MissingNode' but does not"),
         ValidationFailure(node: .editorPlaceholderExpr, message: "could conform to trait 'MissingNode' but does not"),
+        ValidationFailure(node: .editorPlaceholderPattern, message: "could conform to trait 'MissingNode' but does not"),
+        ValidationFailure(node: .editorPlaceholderType, message: "could conform to trait 'MissingNode' but does not"),
```
Comment (kimdv) [inline CodeGeneration/Tests/ValidateSyntaxNodes/ValidateSyntaxNodes.swift:519]: After rebase those should not be needed anymore

**Review by kimdv — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Package.swift [hunk 2]
```diff
@@ -3,36 +3,64 @@
...
+  return ProcessInfo.processInfo.environment[name] != nil
+}
+
+/// Set when building swift-syntax using swift-syntax-dev-utils or in Swift CI in general.
+///
+/// Modifies the build in the following ways
+///  - Enables assertions even in release builds
+///  - Removes the dependency of swift-syntax on os_log
+///  - Enables raw syntax validation (ie. implies `SWIFTSYNTAX_ENABLE_RAWSYNTAX_VALIDATION`)
+///  - Enables alternate token introspection (ie. implies `SWIFTPARSER_ENABLE_ALTERNATE_TOKEN_INTROSPECTION`)
```
Comment (bnbarham) [inline Package.swift:18]: As written, neither of these are true. Both of these are only enabled if the environment variable is set.

File: Package.swift [hunk 3]
```diff
@@ -153,10 +181,11 @@ let package = Package(
 
     .testTarget(
       name: "SwiftSyntaxTest",
-      dependencies: ["_SwiftSyntaxTestSupport", "SwiftSyntax", "SwiftSyntaxBuilder"]
+      dependencies: ["_SwiftSyntaxTestSupport", "SwiftSyntax", "SwiftSyntaxBuilder"],
+      swiftSettings: swiftSyntaxSwiftSettings
```
Comment (bnbarham) [inline Package.swift:185]: Out of interest, was this causing rebuilds of the other targets?

File: SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/Test.swift [hunk 4]
```diff
@@ -32,18 +32,11 @@ struct Test: ParsableCommand, BuildCommand {
...
-
-    if arguments.verbose {
-      swiftpmCallArguments += ["--verbose"]
-    }
 
     try invokeSwiftPM(
       action: "test",
       packageDir: Paths.packageDir,
-      additionalArguments: swiftpmCallArguments,
+      additionalArguments: ["--test-product", "swift-syntaxPackageTests"],
```
Comment (bnbarham) [inline SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/Test.swift:39]: Was `--verbose` not actually used?

File: Package.swift (hunk 3 above)
Comment (ahoppen) [reply Package.swift:185]: Nope, this PR is pure cleanup. Fixing all the rebuilds is
- https://github.com/apple/swift-format/pull/624
- https://github.com/apple/swift-format/pull/630
- https://github.com/apple/swift-stress-tester/pull/251
- https://github.com/apple/swift/pull/68562
- https://github.com/apple/swift-stress-tester/pull/252

**Comment by ahoppen** [comment]
@swift-ci Please test

**Review by bnbarham — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Pull request merged**